File: 485-max-consecutive-ones/max-consecutive-ones.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {

    let maxC = currC = 0;

    for(let i=0;i<nums.length;i++) {

        if(nums[i]==1) currC++;
        else {

            if(currC>maxC) maxC = currC;
            currC = 0;
        }

        
    }
    return Math.max(currC,maxC);
    
};

File: 485-max-consecutive-ones/max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {


        int currC = 0;
        int maxC = 0;

        for(int i=0;i<nums.length;i++) {
            if(nums[i]==1) currC++;
            else {
                if(currC>maxC) maxC = currC;
                currC = 0;
            }
        }

      return Math.max(currC,maxC);
        
        
    }
}